tests/lib/linter.js

```diff
@@ -4041,6 +4067,120 @@ describe("Linter", () => {
                 assert.strictEqual(messages[0].message, errorPrefix + require(parser).expectedError);
             });
 
+            describe("if a parser provides 'visitorKeys'", () => {
+                const types = [];
+                let scopeAnalyzeStub = null;
+                let sourceCode = null;
+
+                before(() => {
+                    scopeAnalyzeStub = sinon.spy(escope, "analyze");
```

Comment:
I would love to see a sinon sandbox used here, to make it easier to restore multiple stubs if we add to this set of tests; but I won't insist on it.